### DISCUSSION TITLE
Equivalence classes

### DISCUSSION THREAD

**Discussion by Ray6283** [COMBINE-lab/salmon#897] [Q&A (bulk specific)]
When I run Salmon with equivalence classes option, I get number of equivalence classes bigger than number of transcripts.  Can it be because of the probabilities and some transcripts may fall into more than on classes? Because, it is supposed to partition the transcripts and one transcript is uniquely in one class  according to equivalence relation defined.
40812 (# transcripts)
47289 (# of Eq classes)
ATMG00010.1
ATMG00030.1
ATMG00040.1
ATMG00050.1
ATMG00060.1

Thanks in advance for your time.

**Comment by rob-p** [comment]
Hi @Ray6283,

The number of transcripts is *not* an upper limit on the number of equivalence classes. In fact, the worst-case limit on the number of equivalence classes is $\min(n, 2^T)$ where $n$ is the number of reads and $T$ is the number of transcripts.  This is even without the *enhanced* equivalence relation of "range-factorized" equivalence classes that salmon uses (which I'll leave out of this particular answer for simplicity). The reason for this that the equivalence relation is defined on *fragments* (i.e. sequenced reads), and it says $f_i \sim f_j$ if and only if $f_i$ and $f_j$ map to the same set of transcripts.  So, clearly there cannot be more equivalence classes than fragments, because at worst each fragment could be in its own class. Likewise, there can't be more equivalence classes than *distinct subsets of transcripts*.  However, there can be many more than the number of transcripts.  Another way to think about this is that the equivalence relation is induced by a *set* of transcripts.  So it's absolutely possible to have an equivalence class (set of reads) that map to transcripts ${A,B,C,D}$, another set that map to ${A, B, C}$ and another set that map just to $\{ A \}$ — a given transcript can appear in the label of many equivalence classes because it appears with a different subset of other transcripts.  So, what you see here is normal and expected.

Best,
Rob

**Comment by Ray6283** [comment]
Hi @rob-p 
Thank you very much.
